recommender/app/controllers/Application.scala

```diff
@@ -51,6 +57,29 @@ object Application extends Controller {
   def hydrateRecommendations(recommendations: List[RecommendationItems]): Future[List[String]] =
     Future.sequence { recommendations map hydrateRecommendation } map { _.flatten }
 
+  def recommendationsFromBrowserId(
+    browserId: String,
+    webPublicationDate: Option[DateRangeFilter],
+    tags: Option[QueryBoost],
+    disableDateFilter: Option[Boolean],
+    pageSize: Option[Int]
+    ) = Action.async {
+    val dateFilter =
+      webPublicationDate
+        .orElse(Some(defaultDateRangeFilter))
+        .filterNot(_ => disableDateFilter.contains(true))
```

Comment:
It took me a while to read this one. Would you mind making it a bit clearer? I suggest something like that:
````scala
disableDateFilter match {
  case Some(true) => None
  case _ => webPublicationDate.getOrElse(defaultDateRangeFilter)
}
````